## Sayanik Mukherjee
## C23027

## Store size prediction
- Predict the Size of store and type of store so that proper prediction can be made based on different feature of product.
- Big Mart sales data contains 6135 rows and 7 feature for 1559 products.

#Install MongoDB

In [ ]:
!apt install mongodb > /dev/null
#!apt install mongodb

In [ ]:
#Local Server Need Not be started for Remote Operations
#!service mongodb start

In [ ]:
from pymongo import MongoClient
import pandas as pd
import dateutil

## Test MongoClient with Remote Server

In [ ]:
# The Password is visible in this cell
# If you wish to hide this then use the next cell
#
rhostName = 'rs0/n1-c2-mongodb-clevercloud-customers.services.clever-cloud.com:27017,n2-c2-mongodb-clevercloud-customers.services.clever-cloud.com:27017'   # place your own credentials here
ruserName = 'uweuwwjhpuuj1u16z0he'      # place your own credentials here
rpassWord = 'EY2yVVAhVqt3XZaAC3dI'      # place your own credentials here
rdbName =  'bueurk2kcxjo3w7'         # place your own credentials here
rMongoURI = 'mongodb://uweuwwjhpuuj1u16z0he:EY2yVVAhVqt3XZaAC3dI@n1-c2-mongodb-clevercloud-customers.services.clever-cloud.com:27017,n2-c2-mongodb-clevercloud-customers.services.clever-cloud.com:27017/bueurk2kcxjo3w7?replicaSet=rs0'         # place your own credentials here

In [ ]:
mongoURI = rMongoURI                                   # for remote server
#mongoURI = "mongodb://localhost:27017/"                 # for local server
client = MongoClient(mongoURI)
client.list_database_names() # ['admin', 'local']

['bueurk2kcxjo3w7']

In [ ]:
#db = client.PraxisDB
db = client.bueurk2kcxjo3w7
db.list_collection_names()

['salse']

In [ ]:
table = db.salse
table.count_documents({})

6135

#MongoDB Exercise <br>
Exercise : https://docs.google.com/document/d/1q0PyJL6EgMG1V5G8mmPre8Ypq9N9dX-PRLixAHiphd8/edit?usp=sharing <br>

Data : restaurant.json https://drive.google.com/file/d/14sYJM2KwIpDRsAAk6YkapMTwLnLuBcb_/view?usp=sharing <br>

For more detailed examples, see https://colab.research.google.com/github/Giffy/MongoDB_PyMongo_Tutorial/blob/master/1_1_First_steps_with_MongoDB_PyMongo.ipynb

##Load Data

In [ ]:
db.drop_collection('salse')

{'$clusterTime': {'clusterTime': Timestamp(1659841939, 1),
  'signature': {'hash': b'\xe6\xb7\x94\xec\xb4\x13\x82f\xfc`\xcc\xa5\xc6\x86\xb1\xff\xb0\x81C\xca',
   'keyId': 7106106537503883265}},
 'nIndexesWas': 1,
 'ns': 'bueurk2kcxjo3w7.salse',
 'ok': 1.0,
 'operationTime': Timestamp(1659841939, 1)}

In [ ]:
# the restaurants.json file is available in my GDrive
# Local Server
#!mongoimport --db PraxisDB --collection restaurants --drop --file restaurants.json
# Remote Server
# mongoimport --host "hostname" --port "port" --db "databasename" --collection "collectionName" --file "filePath"
# the python variables need to be placed in {}
#!mongoimport --host "{rhostName}" --port "27017" --db "{rdbName}" --collection "restaurants"  --drop --file "restaurants.json" -u "{ruserName}" -p "{rpassWord}" --- THIS DOES NOT WORK
!mongoimport --uri "{rMongoURI}"  --collection "salse"  --drop --file "/content/bgmmart.json" --jsonArray

2022-08-07T03:14:24.052+0000	connected to: localhost
2022-08-07T03:14:24.145+0000	dropping: bueurk2kcxjo3w7.salse
2022-08-07T03:14:25.958+0000	imported 6135 documents


##Data Access & Manipulation

In [ ]:
#mongoURI = "mongodb://localhost:27017/"    # local
mongoURI = rMongoURI                        # remote
client = MongoClient(mongoURI)
client.list_database_names()

['bueurk2kcxjo3w7']

In [ ]:
#db = client.PraxisDB
db = client.bueurk2kcxjo3w7
db.list_collection_names()

['salse']

In [ ]:
table = db.salse
table.count_documents({})

6135

In [ ]:
uri = rMongoURI
client = MongoClient( uri )

In [ ]:
client.stats

Database(MongoClient(host=['n2-c2-mongodb-clevercloud-customers.services.clever-cloud.com:27017', 'n1-c2-mongodb-clevercloud-customers.services.clever-cloud.com:27017'], document_class=dict, tz_aware=False, connect=True, replicaset='rs0'), 'stats')

## Installing spark

In [ ]:
!pip3 install -q pyspark

     |████████████████████████████████| 281.3 MB 39 kB/s 
     |████████████████████████████████| 199 kB 48.7 MB/s 


In [ ]:
from pyspark.sql import SparkSession
spark = SparkSession \
    .builder \
    .master("local[*]") \
    .config("spark.mongodb.input.uri", "mongodb://localhost:27017/bueurk2kcxjo3w7.salse") \
    .config("spark.mongodb.output.uri", "mongodb://localhost:27017/bueurk2kcxjo3w7.salse") \
    .config('spark.jars.packages', 'org.mongodb.spark:mongo-spark-connector_2.12:3.0.1')\
    .getOrCreate()

### Accessing the data through Spark

In [ ]:
sparkDF = spark.read.format("mongo").option("uri","mongodb://uweuwwjhpuuj1u16z0he:EY2yVVAhVqt3XZaAC3dI@n1-c2-mongodb-clevercloud-customers.services.clever-cloud.com:27017,n2-c2-mongodb-clevercloud-customers.services.clever-cloud.com:27017/bueurk2kcxjo3w7.salse").load()
sparkDF.show()

+--------+-----------------+---------------+-----------+-------------------------+-----------+--------------------+
|Item_MRP|Item_Outlet_Sales|Item_Visibility|Item_Weight|Outlet_Establishment_Year|Outlet_Size|                 _id|
+--------+-----------------+---------------+-----------+-------------------------+-----------+--------------------+
|249.8092|         3735.138|    0.016047301|        9.3|                     1999|     Medium|{62ef2e104b5e6c80...|
| 48.2692|         443.4228|    0.019278216|       5.92|                     2009|     Medium|{62ef2e104b5e6c80...|
| 141.618|          2097.27|    0.016760075|       17.5|                     1999|     Medium|{62ef2e104b5e6c80...|
| 182.095|           732.38|            0.0|       19.2|                     1998|       null|{62ef2e104b5e6c80...|
| 53.8614|         994.7052|            0.0|       8.93|                     1987|       High|{62ef2e104b5e6c80...|
| 51.4008|         556.6088|            0.0|     10.395|                

In [ ]:
print((sparkDF.count(),len(sparkDF.columns)))

(6135, 7)


The shape of Our big mart data set is 6135 rows and 7 columns.

### Spark schema

In [ ]:
sparkDF.printSchema()

root
 |-- Item_MRP: double (nullable = true)
 |-- Item_Outlet_Sales: double (nullable = true)
 |-- Item_Visibility: double (nullable = true)
 |-- Item_Weight: double (nullable = true)
 |-- Outlet_Establishment_Year: integer (nullable = true)
 |-- Outlet_Size: string (nullable = true)
 |-- _id: struct (nullable = true)
 |    |-- oid: string (nullable = true)



### Descriptive Statistics

In [ ]:
sparkDF.describe().show()

+-------+------------------+------------------+-------------------+------------------+-------------------------+-----------+
|summary|          Item_MRP| Item_Outlet_Sales|    Item_Visibility|       Item_Weight|Outlet_Establishment_Year|Outlet_Size|
+-------+------------------+------------------+-------------------+------------------+-------------------------+-----------+
|  count|              6135|              6135|               6135|              5200|                     6135|       3725|
|   mean|140.61232625916873| 2286.049423504488|0.06408793670285252|12.861068269230488|        1998.127302363488|       null|
| stddev|62.350145630187214|1741.3514252209775|0.04906595890048507| 4.625826003307975|        8.788493760115395|       null|
|    min|             31.29|             33.29|                0.0|             4.555|                     1985|       High|
|    max|          266.8884|        13086.9648|        0.311090379|             21.35|                     2009|     Medium|


## Importing Libraries

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from pyspark.sql.functions import mean
from pyspark.ml.feature import (VectorAssembler,OneHotEncoder, StringIndexer)
from pyspark.ml import Pipeline
from pyspark.ml import PipelineModel
from pyspark.ml.classification import LogisticRegression
from pyspark.ml.classification import RandomForestClassifier
from pyspark.ml.classification import GBTClassifier
from pyspark.ml.evaluation import MulticlassClassificationEvaluator
from pyspark.ml.evaluation import BinaryClassificationEvaluator

In [ ]:
testDF = sparkDF.toPandas()
testDF.info()
testDF.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6135 entries, 0 to 6134
Data columns (total 7 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_MRP                   6135 non-null   float64
 1   Item_Outlet_Sales          6135 non-null   float64
 2   Item_Visibility            6135 non-null   float64
 3   Item_Weight                5200 non-null   float64
 4   Outlet_Establishment_Year  6135 non-null   int32  
 5   Outlet_Size                3725 non-null   object 
 6   _id                        6135 non-null   object 
dtypes: float64(4), int32(1), object(2)
memory usage: 311.7+ KB


Item_MRP  Item_Outlet_Sales  Item_Visibility  Item_Weight  \
count  6135.000000        6135.000000      6135.000000  5200.000000   
mean    140.612326        2286.049424         0.064088    12.861068   
std      62.350146        1741.351425         0.049066     4.625826   
min      31.290000          33.290000         0.000000     4.555000   
25%      93.527800         948.432100         0.026827     8.840000   
50%     142.015400        1874.892800         0.052248    12.600000   
75%     185.375300        3201.499300         0.092540    16.750000   
max     266.888400       13086.964800         0.311090    21.350000   

       Outlet_Establishment_Year  
count                6135.000000  
mean                 1998.127302  
std                     8.788494  
min                  1985.000000  
25%                  1987.000000  
50%                  1999.000000  
75%                  2007.000000  
max                  2009.000000

In [ ]:
testDF.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6135 entries, 0 to 6134
Data columns (total 7 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_MRP                   6135 non-null   float64
 1   Item_Outlet_Sales          6135 non-null   float64
 2   Item_Visibility            6135 non-null   float64
 3   Item_Weight                5200 non-null   float64
 4   Outlet_Establishment_Year  6135 non-null   int32  
 5   Outlet_Size                3725 non-null   object 
 6   _id                        6135 non-null   object 
dtypes: float64(4), int32(1), object(2)
memory usage: 311.7+ KB


We have total 6 columns out of which 5 are numerical type and one is of cetegorical type.

## Data Preprocessing

In [ ]:
#checking null value
sparkDF.toPandas().isnull().sum()

Item_MRP                        0
Item_Outlet_Sales               0
Item_Visibility                 0
Item_Weight                   935
Outlet_Establishment_Year       0
Outlet_Size                  2410
_id                             0
dtype: int64

We have null values in twol columns one is numerical one is categorical we will treat them accordingly.

In [ ]:
# computing missing values for Item-Weight
cmean = sparkDF.select(mean(sparkDF['Item_Weight'])).collect()
mean_weight = cmean[0][0]
sparkDF2 = sparkDF.na.fill(mean_weight,['Item_Weight'])

In [ ]:
# Computing ,issing values for outlet size
sparkDF2 = sparkDF2.na.fill('No Info', subset=['Outlet_Size'])

In [ ]:
#Again checking null value
sparkDF2.toPandas().isnull().sum()

Item_MRP                     0
Item_Outlet_Sales            0
Item_Visibility              0
Item_Weight                  0
Outlet_Establishment_Year    0
Outlet_Size                  0
_id                          0
dtype: int64

In [ ]:
sparkDF2.dtypes

[('Item_MRP', 'double'),
 ('Item_Outlet_Sales', 'double'),
 ('Item_Visibility', 'double'),
 ('Item_Weight', 'double'),
 ('Outlet_Establishment_Year', 'int'),
 ('Outlet_Size', 'string'),
 ('_id', 'struct<oid:string>')]

## String Indexer

In [ ]:
Outlet_Size_Indexer = StringIndexer(inputCol="Outlet_Size", outputCol="Outlet_SizeIndex")

In [ ]:
Outlet_Size_Indexer.fit(sparkDF2).transform(sparkDF2).show(5)

+--------+-----------------+---------------+-----------+-------------------------+-----------+--------------------+----------------+
|Item_MRP|Item_Outlet_Sales|Item_Visibility|Item_Weight|Outlet_Establishment_Year|Outlet_Size|                 _id|Outlet_SizeIndex|
+--------+-----------------+---------------+-----------+-------------------------+-----------+--------------------+----------------+
|249.8092|         3735.138|    0.016047301|        9.3|                     1999|     Medium|{62ef2e104b5e6c80...|             0.0|
| 48.2692|         443.4228|    0.019278216|       5.92|                     2009|     Medium|{62ef2e104b5e6c80...|             0.0|
| 141.618|          2097.27|    0.016760075|       17.5|                     1999|     Medium|{62ef2e104b5e6c80...|             0.0|
| 182.095|           732.38|            0.0|       19.2|                     1998|    No Info|{62ef2e104b5e6c80...|             1.0|
| 53.8614|         994.7052|            0.0|       8.93|             

## One Hot Encoder

In [ ]:
OHE_Outlet_Size = OneHotEncoder(inputCols=["Outlet_SizeIndex"], outputCols=["Outlet_SizeVec"])

In [ ]:
OHE_Outlet_Size.fit(Outlet_Size_Indexer.fit(sparkDF2).transform(sparkDF2)).transform(Outlet_Size_Indexer.fit(sparkDF2).transform(sparkDF2)).show(5)

+--------+-----------------+---------------+-----------+-------------------------+-----------+--------------------+----------------+--------------+
|Item_MRP|Item_Outlet_Sales|Item_Visibility|Item_Weight|Outlet_Establishment_Year|Outlet_Size|                 _id|Outlet_SizeIndex|Outlet_SizeVec|
+--------+-----------------+---------------+-----------+-------------------------+-----------+--------------------+----------------+--------------+
|249.8092|         3735.138|    0.016047301|        9.3|                     1999|     Medium|{62ef2e104b5e6c80...|             0.0| (2,[0],[1.0])|
| 48.2692|         443.4228|    0.019278216|       5.92|                     2009|     Medium|{62ef2e104b5e6c80...|             0.0| (2,[0],[1.0])|
| 141.618|          2097.27|    0.016760075|       17.5|                     1999|     Medium|{62ef2e104b5e6c80...|             0.0| (2,[0],[1.0])|
| 182.095|           732.38|            0.0|       19.2|                     1998|    No Info|{62ef2e104b5e6c80.

## Assembler

In [ ]:
F_assembler = VectorAssembler(inputCols=['Item_MRP',
 'Item_Weight','Item_Visibility',
 'Outlet_Establishment_Year'],outputCol='features')

In [ ]:
F_assembler

VectorAssembler_e2e426429f92

# Train-Test Split

In [ ]:
# splitting training and validation data
train_mart,val_mart = sparkDF2.randomSplit([0.7,0.3])
print(train_mart.count())
print(val_mart.count())

4269
1866


#Apply Different Techniques

### Making Pipeline

In [ ]:
basePipe = Pipeline(stages=[Outlet_Size_Indexer,OHE_Outlet_Size, F_assembler])
basePipe.fit(sparkDF2).transform(sparkDF2).show(5)

+--------+-----------------+---------------+-----------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+
|Item_MRP|Item_Outlet_Sales|Item_Visibility|Item_Weight|Outlet_Establishment_Year|Outlet_Size|                 _id|Outlet_SizeIndex|Outlet_SizeVec|            features|
+--------+-----------------+---------------+-----------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+
|249.8092|         3735.138|    0.016047301|        9.3|                     1999|     Medium|{62ef2e104b5e6c80...|             0.0| (2,[0],[1.0])|[249.8092,9.3,0.0...|
| 48.2692|         443.4228|    0.019278216|       5.92|                     2009|     Medium|{62ef2e104b5e6c80...|             0.0| (2,[0],[1.0])|[48.2692,5.92,0.0...|
| 141.618|          2097.27|    0.016760075|       17.5|                     1999|     Medium|{62ef2e104b5e6c80...|             0.0| (2,[0],[1.0])|[141.618

## Logistic Regression Model Pipeline

In [ ]:
from pyspark.ml.classification import LogisticRegression

In [ ]:
lr = LogisticRegression(labelCol='Outlet_SizeIndex',featuresCol='features',maxIter=5)
lr_pipeline = Pipeline(stages=[basePipe,lr])                    # Appending a stage to a pipeline
# training model pipeline with data
lr_model = lr_pipeline.fit(train_mart)
lr_predictions=lr_model.transform(val_mart)

In [ ]:
lr_predictions.show(5)

+--------+-----------------+---------------+------------------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+--------------------+--------------------+----------+
|Item_MRP|Item_Outlet_Sales|Item_Visibility|       Item_Weight|Outlet_Establishment_Year|Outlet_Size|                 _id|Outlet_SizeIndex|Outlet_SizeVec|            features|       rawPrediction|         probability|prediction|
+--------+-----------------+---------------+------------------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+--------------------+--------------------+----------+
|   31.49|           466.06|    0.041851461|               9.5|                     1987|       High|{62ef2e114b5e6c80...|             2.0|     (2,[],[])|[31.49,9.5,0.0418...|[0.70297268674228...|[0.54158187941311...|       0.0|
|   31.89|           366.19|    0.042056944|               9.5|                     

In [ ]:
acc_evaluator = MulticlassClassificationEvaluator(labelCol="Outlet_SizeIndex", predictionCol="prediction", metricName="accuracy")
evaluator = BinaryClassificationEvaluator(labelCol='Outlet_SizeIndex')

In [ ]:
lr_acc=acc_evaluator.evaluate(lr_predictions)
print('A Logistic Regression algorithm had an accuracy of: {0:2.2f}%'.format(lr_acc*200))
#print(round(lr_acc,3), 'is the accuray of the LR pipeline')
# lr_auroc = evaluator.evaluate(lr_predictions, {evaluator.metricName: "areaUnderROC"})
# #auprc = evaluator.evaluate(predictions, {evaluator.metricName: "areaUnderPR"})
# print("Area under ROC Curve: {:.4f}".format(lr_auroc))
# #print("Area under PR Curve: {:.4f}".format(auprc)


A Logistic Regression algorithm had an accuracy of: 70.74%


Logistic Regression is giving us very least accuracu 70.74 so we have to look for aother algoritham also

## Naive Bayes Pipeline

In [ ]:
from pyspark.ml.classification import NaiveBayes

In [ ]:
Nv = NaiveBayes(featuresCol = 'features', labelCol = 'Outlet_SizeIndex')
Nv_pipeline = Pipeline(stages=[basePipe, Nv])
# training model pipeline with data
Nv_model = Nv_pipeline.fit(train_mart)
Nv_predictions=Nv_model.transform(val_mart)

In [ ]:
Nv_predictions.show(5)

+--------+-----------------+---------------+------------------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+--------------------+--------------------+----------+
|Item_MRP|Item_Outlet_Sales|Item_Visibility|       Item_Weight|Outlet_Establishment_Year|Outlet_Size|                 _id|Outlet_SizeIndex|Outlet_SizeVec|            features|       rawPrediction|         probability|prediction|
+--------+-----------------+---------------+------------------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+--------------------+--------------------+----------+
|   31.49|           466.06|    0.041851461|               9.5|                     1987|       High|{62ef2e114b5e6c80...|             2.0|     (2,[],[])|[31.49,9.5,0.0418...|[-282.40877280979...|[0.62836550077062...|       0.0|
|   31.89|           366.19|    0.042056944|               9.5|                     

In [ ]:
Nv_acc=acc_evaluator.evaluate(Nv_predictions)
# lr_auroc = evaluator.evaluate(lr_predictions, {evaluator.metricName: "areaUnderROC"})
# #auprc = evaluator.evaluate(predictions, {evaluator.metricName: "areaUnderPR"})
print('A Naive algorithm had an accuracy of: {0:2.2f}%'.format(Nv_acc*200))
# print("Area under ROC Curve: {:.4f}".format(lr_auroc))
# #print("Area under PR Curve: {:.4f}".format(auprc))

A Naive algorithm had an accuracy of: 78.46%


accuracy of naive algooritham is 78.64%

##  Random Forest Pipeline

In [ ]:
rf = RandomForestClassifier(featuresCol = 'features', labelCol = 'Outlet_SizeIndex')
#rf_pipeline = Pipeline(stages=[GenderIndexer, SmokeHistIndexer, OHE_Gender_Smoke, F_assembler, rf])
rf_pipeline = Pipeline(stages=[basePipe, rf])
# training model pipeline with data
rf_model = rf_pipeline.fit(train_mart)
rf_predictions=rf_model.transform(val_mart)

In [ ]:
rf_predictions.show(5)

+--------+-----------------+---------------+------------------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+--------------------+--------------------+----------+
|Item_MRP|Item_Outlet_Sales|Item_Visibility|       Item_Weight|Outlet_Establishment_Year|Outlet_Size|                 _id|Outlet_SizeIndex|Outlet_SizeVec|            features|       rawPrediction|         probability|prediction|
+--------+-----------------+---------------+------------------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+--------------------+--------------------+----------+
|   31.49|           466.06|    0.041851461|               9.5|                     1987|       High|{62ef2e114b5e6c80...|             2.0|     (2,[],[])|[31.49,9.5,0.0418...|[1.03818369453044...|[0.05190918472652...|       2.0|
|   31.89|           366.19|    0.042056944|               9.5|                     

In [ ]:
rf_acc=acc_evaluator.evaluate(rf_predictions)
print('A Random Forest algorithm had an accuracy of: {0:2.2f}%'.format(rf_acc*100))

A Random Forest algorithm had an accuracy of: 95.50%


Random Forest is giving us accuracy of 95.50%

## Decision Tree Pipeline

In [ ]:
from pyspark.ml.classification import DecisionTreeClassifier

In [ ]:
dt = DecisionTreeClassifier(featuresCol = 'features', labelCol = 'Outlet_SizeIndex')
#rf_pipeline = Pipeline(stages=[GenderIndexer, SmokeHistIndexer, OHE_Gender_Smoke, F_assembler, rf])
dt_pipeline = Pipeline(stages=[basePipe, dt])
# training model pipeline with data
dt_model = rf_pipeline.fit(train_mart)
dt_predictions=rf_model.transform(val_mart)

In [ ]:
dt_predictions.show(5)

+--------+-----------------+---------------+------------------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+--------------------+--------------------+----------+
|Item_MRP|Item_Outlet_Sales|Item_Visibility|       Item_Weight|Outlet_Establishment_Year|Outlet_Size|                 _id|Outlet_SizeIndex|Outlet_SizeVec|            features|       rawPrediction|         probability|prediction|
+--------+-----------------+---------------+------------------+-------------------------+-----------+--------------------+----------------+--------------+--------------------+--------------------+--------------------+----------+
|   31.49|           466.06|    0.041851461|               9.5|                     1987|       High|{62ef2e114b5e6c80...|             2.0|     (2,[],[])|[31.49,9.5,0.0418...|[1.03818369453044...|[0.05190918472652...|       2.0|
|   31.89|           366.19|    0.042056944|               9.5|                     

In [ ]:
dt_acc=acc_evaluator.evaluate(dt_predictions)
print('A Decision Tree algorithm had an accuracy of: {0:2.2f}%'.format(dt_acc*100*))

A Decision Tree algorithm had an accuracy of: 95.50%


# Comparison of LR,Naive Bayes, RF , DT

In [ ]:
print(round(lr_acc,3), 'is the accuray of the LR pipeline')
print(round(Nv_acc,3), 'is the accuray of the Nv pipeline')
print(round(rf_acc,3), 'is the accuray of the RF pipeline')
#print(round(dt_acc,3), 'is the accuray of the DT pipeline')


0.354 is the accuray of the LR pipeline
0.392 is the accuray of the Nv pipeline
0.955 is the accuray of the RF pipeline


- Rndom forest is giving us best accuracy so we will consider this as our prediction Model
- In future for further improvement hyperparameter optimization can be done to improve accuracy
